contracts/core/Amm.sol

```diff
@@ -123,30 +138,83 @@ contract Amm is IAmm, LiquidityERC20, Reentrant {
             uint256 liquidity
         )
     {
+        // only router can burn liquidity 
+        require(IConfig(config).routerMap(msg.sender), "Amm.mint: FORBIDDEN");
         (uint112 _baseReserve, uint112 _quoteReserve, ) = getReserves(); // gas savings
         liquidity = balanceOf[address(this)];
 
         // get real baseReserve
-        int256 baseTokenOfNetPosition = IMargin(margin).netPosition();
-        int256 realBaseReserveSigned = int256(uint256(_baseReserve)) + baseTokenOfNetPosition;
-        uint256 realBaseReserve = uint256(realBaseReserveSigned);
-
+        uint256 realBaseReserve = getRealBaseReserve();
+     
+        // calculate the fee
         bool feeOn = _mintFee(_baseReserve, _quoteReserve);
+
         uint256 _totalSupply = totalSupply; // gas savings, must be defined here since totalSupply can update in _mintFee
 
-        baseAmount = (liquidity * realBaseReserve) / _totalSupply; // using balances ensures pro-rata distribution
+        baseAmount = (liquidity * realBaseReserve) / _totalSupply; 
         // quoteAmount = (liquidity * _quoteReserve) / _totalSupply; // using balances ensures pro-rata distribution
         quoteAmount = (baseAmount * _quoteReserve) / _baseReserve;
 
+
         require(baseAmount > 0 && quoteAmount > 0, "Amm.burn: INSUFFICIENT_LIQUIDITY_BURNED");
+
+         // gurantee the netpostion close in a tolerant sliappage after remove liquidity
+        int256 quoteTokenOfNetPosition = IMargin(margin).netPosition();
+    
+        uint256 lpWithdrawThreshold = IConfig(config).lpWithdrawThreshold();
+  
+        require(
+            quoteTokenOfNetPosition.abs() * 100 < (_quoteReserve - quoteAmount) * lpWithdrawThreshold,
+            "Amm.burn: TOO_LARGE_LIQUIDITY_WITHDRAW"
+        );
+
+   
+        require(
+            (_baseReserve - baseAmount) * _quoteReserve * 999 <= (_quoteReserve - quoteAmount) * _baseReserve * 1000,
+            "Amm.burn: PRICE_BEFORE_AND_AFTER_MUST_BE_THE_SAME"
+        );
+        require(
```

Comment:
there's no reason to have this be 1000 & 999 in one check and 1001 and 1000 in the other, right?